kubeflow/jupyter/sync-notebook.jsonnet

```diff
@@ -3,6 +3,7 @@
 // - Service
 // - Pod
 function(request) {
+  local util = import "util.libsonnet",
```

Comment:
should be 
```
  local util = import "kubeflow/common/util.libsonnet",
```

Reply:
If i use full path i get the below error
```failed to sync Notebook "resnet50/training": sync hook failed for Notebook resnet50/training: sync hook failed: remote error: RUNTIME ERROR: Couldn't open import "kubeflow/common/util.libsonnet": No match locally or in the Jsonnet library paths.```

Reply:
Also, because of this one of the test fails. Is there a lib path where i can add it?

Reply:
test_jsonnet.py is manually evaluating jsonnet files and util.libsonnet is added to configmap and might not be available to test. Any Ideas?

Reply:
I guess copying util.libsonnet to kubeflow/jupyter/tests might help.

Reply:
Can you just add the appropriate directory for util.libsonnet to the jsonnet path to get the jsonnet test to pass?

We set the path here
https://github.com/kubeflow/kubeflow/blob/master/testing/workflows/components/unit_tests.jsonnet#L126

Reply:
It didn't help. Fails with same error

Reply:
I have even tried copying util.libsonnet to workflows/lib/v1.7.0. When i run locally it returns true.

jsonnet eval kubeflow/jupyter/tests/sync-notebook_test.jsonnet --jpath testing/workflows/lib/v1.7.0/

Reply:
Here's my conjectures about the issues with the import and the test failure.

sync-notebook.jsonnet is the function executed by the lambda metacontroller. So when running in an actual deployment "util.libsonnet" is made available via the ConfigMap. The net result is that it should be imported as "util.libsonnet" when running on the controller.

So in the test we need to simulate that by adding kubeflow/common to the jpath.

Reply:
This should work.
 jsonnet eval ~/go/src/github.com/kubeflow/kubeflow/jupyter/tests/sync-notebook_test.jsonnet --jpath  ~/go/src/github.com/kubeflow/kubeflow/testing/workflows/lib/v1.7.0/  --jpath  ~/go/src/github.com/kubeflow/kubeflow/common

Reply:
To summarize the discussion in slack:

Adding kubeflow/common to the jsonnet path will inevitably lead people to import util as "util.libsonnet" in their ksonnet files which will work when evaluated by jsonnet eval in the tests but not in an actual ks app because of differences in how the path is set.

As a result we decided that the better short term solution was just to duplicate util.libsonnet in kubeflow/jupyter.

My conjecture is that we will end up reimplementing the controller in go pretty soon and we can get rid of the code duplication.